Assets/Labrys/FeatureEditor/FeatureAsset.cs

```diff
@@ -0,0 +1,372 @@
+﻿using Labrys.Generation;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Labrys.FeatureEditor
+{
+	/// <summary>
+	/// A file-backed representation of a Feature for use with the Feature Editor
+	/// </summary>
+	[CreateAssetMenu(menuName = "Labrys/Feature", fileName = "New Feature")]
+	public class FeatureAsset : ScriptableObject, ISerializationCallbackReceiver
+	{
+		[Serializable]
+		public class Section
+		{
+			public static bool IsValidPosition(Vector2Int gridPosition)
+			{
+				return gridPosition.x % GRID_DENSITY == 0 && gridPosition.y % GRID_DENSITY == 0;
+			}
+
+			public string Variant { get; set; }
+		}
+
+		[Serializable]
+		public class Link
```

Comment:
also kinda same arguments for static class but a bit weaker here since Link is used

Reply:
Also same argument as for Section class.